### ISSUE TITLE
Install Julia 1.9.2

### ISSUE THREAD

**Issue by dphow** [NCAR/spack-derecho#7]
See https://julialang.org/downloads/ and https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/julia/package.py

**Comment by vanderwb** [comment]
Cross-posting from Slack - this is ready for testing in the build environment.

**Comment by vanderwb** [comment]
@dphow - have you gotten any feedback from Tomas on how this version is working out?

**Comment by dphow** [comment]
Using
```shell
. /glade/work/csgteam/spack-deployments/derecho/23.06/envs/build/bin/use_modules
module load julia/1.9.2
```

not working so far with some compiler errors. See https://ithelp.ucar.edu/browse/RC-19432

To note, the error can be replicated in Julia by simply trying to install the package he wants `import Pkg; Pkg.add("DrWatson")`. I get similar errors installing other packages that perform pre-compilation. An errors.txt file is included in Tomas' ticket.

Some other errors include text such as `IOError: could not spawn setenv(...)` which then ends with `no such file or directory (ENOENT)`. Thus, I would presume some folder being set is not being found.

**Comment by vanderwb** [comment]
@dphow - I've remove the broken Julia test build in the `ncarenv/23.06` stack. However, I believe I've found the problem and put together a fix (498804b), which is now in a **published** `julia/1.9.2` module in the new `ncarenv/23.09` stack. Let me know if this resolves things for you and Tomas.

I'm not in love with this fix and I think the build should be doing something smarter to point to LLVM, but that's an investigation for another time I think!
